spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/ExportUtil.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec.bootstrap.data.util;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * Utility methods for managing manifest exports.
+ */
+public final class ExportUtil {
+    private static final String EXPORT_NAME = "consumer_export.zip";
+
+    private ExportUtil() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Retrieves the export zip file in the manifest.
+     *
+     * @param manifest compressed manifest to retrieve the export from.
+     * @return a compressed {@link ZipFile} of the export.
+     * @throws IllegalArgumentException if manifest is null.
+     * @throws IOException if unable to create a temporary export zip file.
+     */
+    public static ZipFile getExportArchive(File manifest) throws IOException {
+        if (manifest == null) {
+            throw new IllegalArgumentException("Manifest is null.");
+        }
+
+        try (ZipFile zipFile = new ZipFile(manifest)) {
+            ZipEntry entry = zipFile.getEntry(EXPORT_NAME);
+            File tmp = File.createTempFile("export", ".zip");
+            tmp.deleteOnExit();
+            try (InputStream istream = zipFile.getInputStream(entry);
+                FileOutputStream ostream = new FileOutputStream(tmp)) {
+                istream.transferTo(ostream);
+            }
+
+            return new ZipFile(tmp);
+        }
+    }
+
+    /**
+     * Retrieves all {@link ZipEntry}s in a provided path inside of the provided {@link ZipFile}.
+     *
+     * @param path location of the directory to retrieve zip entries from.
+     * @param zipFile the zip file system to retrieve the zip entries from.
+     * @return the zip entries in the provided zip file and target directory.
+     */
+    public static List<ZipEntry> getFilesInDirectory(String path, ZipFile zipFile) {
+        List<ZipEntry> foundFiles = new ArrayList<>();
+        Enumeration<? extends ZipEntry> zipEntries = zipFile.entries();
```

Comment:
I should probably clarify before it's too late: when comparing the name to a path, if we're going to do it correctly, we should also verify that the match we're making is on a proper directory separator so we don't get into this situation:

path: "/my/target/dir"
"/my/target/dir" - exact match
"/my/target/dir/file" - also a match (starts with path)
"/my/target/dirname/file" - shouldn't match but does (starts with path, but not in the way we want)

The inverse is true of filenames:
filename: "myfile.txt"
"/my/target/dir/myfile.txt" - exact match
"/my/target/dir/thisismyfile.txt" - also matches but shouldn't

This is less of a concern with ZipEntrys since they appear to use the FQ name for everything so we're likely to pass in the FQ path as well, but it's something to keep in mind.

Reply:
Updated to use a steam instead of enumeration, and I added a helper method isEntryInDirectory that verifies that the file starts with the target directory. To address the sub directory issue I check if there is an additional '/' in the path after the beginning target directory substring. I'm not sure if there is a better way of validating the file is not in sub directory of the target path.